client/model/src/main/java/io/smallrye/graphql/client/model/MethodKey.java

```diff
@@ -4,9 +4,9 @@
 import java.util.Objects;
 
 public class MethodKey {
-    private Class<?> declaringClass;
-    private String methodName;
-    private Class<?>[] parameterTypes;
+    private final Class<?> declaringClass;
```

Comment:
@jmartisk Do you think that this will work with the bytecode recorder? (removing set methods)

Reply:
It definitely must not be final and must have getters and setters to work with Quarkus bytecode recorder

Reply:
Oh yeah, I was planning to add some Javadoc in the model classes (ClientModels, ClientModel, MethodKey), so that's partially my fault for not explaining the reason for having the set methods.

Reply:
I just reverted that; and added javadoc and Deprecated annotations